proposals/0100-read-only-reserved-accounts.md

```diff
@@ -0,0 +1,140 @@
+the Solana core protocol that should not be write locked by a transaction. This
+excludes the incinerator account which is designed to be write locked.
+
+## Detailed Design
+
+To determine the post-demotion set of write lock requested keys for a
+transaction, each transaction write locked account key is checked against a set
+of reserved keys maintained by the validator for a given block.
+
+The set of reserved keys will be initialized as the full set of sysvar and
+builtin program keys currently tracked in static lists in the Solana Labs
+validator implementation:
+
+sysvar_ids:
+
+- SysvarC1ock11111111111111111111111111111111
+- SysvarEpochSchedu1e111111111111111111111111
+- SysvarFees111111111111111111111111111111111
+- SysvarRecentB1ockHashes11111111111111111111
+- SysvarRent111111111111111111111111111111111
+- SysvarRewards111111111111111111111111111111
+- SysvarS1otHashes111111111111111111111111111
+- SysvarS1otHistory11111111111111111111111111
+- SysvarStakeHistory1111111111111111111111111
+- Sysvar1nstructions1111111111111111111111111
+
+builtin_program_ids:
+
+- Config1111111111111111111111111111111111111
+- Feature111111111111111111111111111111111111
+- NativeLoader1111111111111111111111111111111
+- Stake11111111111111111111111111111111111111
+- StakeConfig11111111111111111111111111111111
+- Vote111111111111111111111111111111111111111
+- 11111111111111111111111111111111
+- BPFLoader1111111111111111111111111111111111
+- BPFLoader2111111111111111111111111111111111
+- BPFLoaderUpgradeab1e11111111111111111111111
+
+The set of reserved keys may only be modified via feature gate activation. On
+epoch boundaries, validator implementations should add or remove reserved keys
+as dictated by feature gated code. The first feature gate activated modification
+will likely include the following keys:
+
+new_reserved_keys:
+
+- AddressLookupTab1e1111111111111111111111111
+- ComputeBudget111111111111111111111111111111
+- Ed25519SigVerify111111111111111111111111111
+- KeccakSecp256k11111111111111111111111111111
+- LoaderV411111111111111111111111111111111111
+- Sysvar1111111111111111111111111111111111111
+- SysvarEpochRewards1111111111111111111111111
+- SysvarLastRestartS1ot1111111111111111111111
+- ZkTokenProof1111111111111111111111111111111
+
+For validator operations not subject to consensus like RPC services, ledger
+history storage, and debugging tools which don't have a block context for
+fetching the set of reserved accounts, a static list can be maintained and
+updated without requiring feature gates.
+
+## Impact
+
+Impact should be negligible. dApp developers don't need to change how they build
+transactions due to the developer friendly nature of the write lock demotion
+feature.
+
+## Security Considerations
+
+The main consideration is making sure that all `is_writable` checks are
+consistent before and after implementing this proposal to avoid breaking
+consensus.
```

Comment:
i'd add a note about this not scaling to too many reserved addresses. should we decide that we need to provide this behavior to a large number of accounts in the future, we should probably actually flip a bit in the accounts' metadata

Reply:
Agreed. In the future I would like the account metadata to designate a write lock authority.